packages/core/src/utils/web3wrapper.ts

```diff
@@ -71,6 +71,14 @@ export class Web3Wrapper {
     return this.web3.version.network;
   }
 
+  public async getTimestampForBlock(blockNumber: number): Promise<BigNumber> {
+    return new Promise<BigNumber>((resolve, reject) => {
```

Comment:
I think this is too narrow, since they can just get the timestamp after getting block.

How about:

```typescript
public async getBlock(blockNumber: number): Promise<Web3.Block> {
  const getBlockAsync = promisify<Web3.Block>(this.web3.eth.getBlock, this.web3.eth);
  return getBlockAsync(blockNumber);
}
```